### DISCUSSION TITLE
Selecting a substring, not the whole log line

### DISCUSSION THREAD

**Discussion by abrothman-vacatia** [Kittyfisto/Tailviewer#377] [Ideas]
I frequently need to select only a single word of the log line - an ID to look up in the database. Right now, I can only select and copy the whole line, so I need to paste it into a text editor to get the substring I need.

It would be great if a key modifier or double-click would allow the user to select a substring on the line.

**Comment by mark-moran-wne** [comment]
Regex capture groups could be used as an ad-hoc syntax highlighting/scraping solution.

Ex: `Process is 1% complete (1/250)`
Regex: `Process is (.*)% complete \((.*)/(.*)\)`

Result: Process is `1`% complete (`1`/`250`)

Colors could be assigned to each capture group, or only those values displayed in a table format. 

I wonder if it could be done as a plugin?

**Reply by abrothman-vacatia** [reply]
That really doesn't suit the use case. What I'm grabbing is different every time.

I just want to be able to do this (video link):

https://www.dropbox.com/scl/fi/em6mejy3ck7s3xkuhxx5h/notepad-_nHzJSU3Cdw.mp4?rlkey=yercrnzc26gqfgn2olxtgf6xg&dl=0

**Reply by mark-moran-wne** [reply]
`ERROR: (.*) is not allowed.*$`
<img width="1699" alt="image" src="https://github.com/user-attachments/assets/55f07d92-6055-4305-b6bb-42c5b1a61364">

**Reply by mark-moran-wne** [reply]
I guess I'm thinking more about capturing certain items and you just want to select text adhoc.

**Reply by abrothman-vacatia** [reply]
Yes. I understand that writing a regex every time I want to copy text out of the log is a workaround. However, it's not a USEFUL workaround. It turns a one-second task into a - best case - 20 second task.

If I were opening the log file in Notepad, the flower of 1983 technology, I could select the text. Am I crazy for wanting the same obvious functionality for a log viewer in 2024?
